### ISSUE TITLE
Loss about vae and diffusion

### ISSUE THREAD

**Issue by 980202006** [Stability-AI/stable-audio-tools#63]
I am training VAE and stable audio2 models from scratch, how much will VAE and Diffusion loss reach?My current VAE loss is about 0.4, and diffusion’s mse loss is 0.53.

**Comment by BingliangLi** [comment]
You VAE loss is really low, may I ask how many steps do you train the VAE? I'm currently training the VAE for about 300000 steps and the loss is about 3.5. I'm using a quite large dataset.

**Comment by 980202006** [comment]
Sorry, I remembered it wrong. VAE loss is about 1.0, but the effect is still not good enough.
![image](https://github.com/Stability-AI/stable-audio-tools/assets/24452502/55d04fdd-a996-47d0-a277-de9c783ba22d)
Can you tell me about your data volume and distribution? Maybe it's a coding problem?

**Comment by BingliangLi** [comment]
I'm using the stable_audio_2_0_vae.json config, i didn't change the model. I think maybe your dataset is not large enough? I'm using Audioset + VGGSound + mtg-jamendo + BBCSoundEffect +CommonVoice + free_to_use_sound and a bunch of other smaller dataset, here is my sample:

https://github.com/Stability-AI/stable-audio-tools/assets/49446651/267ec9b9-3b93-474f-ae75-3a30963491d6

**Comment by 980202006** [comment]
It is true that your data set is more complex and the loss should be larger. Is there a demo of the reconstruction effect?

**Comment by 980202006** [comment]
Yes! This reconstruction works very well. What is the loss of the diffusion model you trained?

**Comment by BingliangLi** [comment]
I will start to train the diffusion model in the next few days, will update my status in this issue :)

**Comment by 980202006** [comment]
Waiting for your good news!

**Comment by nateraw** [comment]
Thank you for sharing @BingliangLi  ❤️

**Comment by PeiwenSun2000** [comment]
Would you kindly share your model config and dataset config for beginners' reference?
I would really appreciate it!!

**Comment by xianshenglee** [comment]
> Sorry, I remembered it wrong. VAE loss is about 1.0, but the effect is still not good enough. ![image](https://private-user-images.githubusercontent.com/24452502/328152970-55d04fdd-a996-47d0-a277-de9c783ba22d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxMjYxNjYsIm5iZiI6MTcxNzEyNTg2NiwicGF0aCI6Ii8yNDQ1MjUwMi8zMjgxNTI5NzAtNTVkMDRmZGQtYTk5Ni00N2QwLWEyNzctZGU5Yzc4M2JhMjJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDAzMjQyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwNDBjYTQyNmUyOGY1YTg3NmEyYjc4ZDJmZmVmNWFlNTczODNmMDM3MjA1MjAzNWM2YWYxMmE1YzYwZDlmYjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.B2DdqOkByXa5ugMUrys9LXG76SYRO_gm8Bx7inEfH-Q) Can you tell me about your data volume and distribution? Maybe it's a coding problem?

my diffusion's mse loss is 0.8 and I have the same concern with you... could you please also share your diffusion loss curve figure ? Thank you.

**Comment by Alidaling** [comment]
> 我使用的是 stable_audio_2_0_vae.json 配置，我没有更改模型。我想也许你的数据集不够大？我使用的是 Audioset + VGGSound + mtg-jamendo + BBCSoundEffect +CommonVoice + free_to_use_sound 和一堆其他较小的数据集，这是我的示例：
> 
>  侦察_00326001.mp4

This reconstruction works very well，How many GPUs did you use for training, and what was the batch size?

**Comment by MP-HOI** [comment]
> > 我使用的是 stable_audio_2_0_vae.json 配置，我没有更改模型。我想也许你的数据集不够大？我使用的是 Audioset + VGGSound + mtg-jamendo + BBCSoundEffect +CommonVoice + free_to_use_sound 和一堆其他较小的数据集，这是我的示例：
> > 侦察_00326001.mp4
> 
> This reconstruction works very well，How many GPUs did you use for training, and what was the batch size?

I trained the model for 1000000 stpes with 8 A100 80G, with batch size set to 16, however the recon result is good enough around 500000 steps. I haven't start to train the diffusion yet, but I will do it before july.

**Comment by BingliangLi** [comment]
> > 我使用的是 stable_audio_2_0_vae.json 配置，我没有更改模型。我想也许你的数据集不够大？我使用的是 Audioset + VGGSound + mtg-jamendo + BBCSoundEffect +CommonVoice + free_to_use_sound 和一堆其他较小的数据集，这是我的示例：
> > 侦察_00326001.mp4
> 
> This reconstruction works very well，How many GPUs did you use for training, and what was the batch size?

I trained the model for 1000000 stpes with 8 A100 80G, with batch size set to 16, however the recon result is good enough around 500000 steps. I haven't start to train the diffusion yet, but I will do it before July.

**Comment by cvillela** [comment]
hey there @BingliangLi , did you endup training the diffusion model? Great results! I would love to know what parameters you used
